packages/participants-svc/src/domain/participant_agg.ts

```diff
@@ -1583,17 +1584,40 @@ export class ParticipantAggregate {
                 `Cannot find a hub's assets account for currency: ${fundsMov.currencyCode}`
             );
         }
-        const positionAccount = participant.participantAccounts.find(
+        const settlementAccount = participant.participantAccounts.find(
             (value: IParticipantAccount) =>
                 value.currencyCode === fundsMov.currencyCode &&
                 value.type === "SETTLEMENT"
         );
-        if (!positionAccount) {
+        if (!settlementAccount) {
             throw new AccountNotFoundError(
                 `Cannot find a participant's position account for currency: ${fundsMov.currencyCode}`
             );
         }
 
+        // Check if enough balance in settlement account for withdrawal
+        if (fundsMov.direction === ParticipantFundsMovementDirections.FUNDS_WITHDRAWAL) {
+            // Get the account from account and balance adapter
+            const updatedSettlementAcc = await this._accBal.getAccount(settlementAccount.id);
+            if (!updatedSettlementAcc) {
+                throw new AccountNotFoundError(
+                    `Could not get settlement account from accountsAndBalances adapter for participant id: ${participant.id}`
+                );
+            }
+            
+            if (!updatedSettlementAcc.balance) {
```

Comment:
I think this is an unsafe way of checking for a balance. Note that js will return true in some circumstances you do not wish to proceed here e.g. if balance is a string, or a negative integer. I would suggest a little more rigour here in deciding if the balance property is "correct" at this point.

Reply:
Got it and will take care of it. Thanks.

Reply:
This check is to conclude an account does not exist or has a bad balance value and break early, I would say this is being cautious.
Anyway, @Zwe333 please try to convert to number before processing with the comparison.